client/ayon_silhouette/api/plugin.py

```diff
@@ -0,0 +1,191 @@
+        shared_data(Dict[str, Any]): Shared data.
+
+    """
+    if shared_data.get('silhouette_cached_instances') is None:
+        shared_data["silhouette_cached_instances"] = cache = {}
+
+        session = fx.activeSession()
+        if not session:
+            return
+
+        instance_ids = {AYON_INSTANCE_ID, AVALON_INSTANCE_ID}
+        for node in session.nodes:
+            data = lib.read(node)
+            if not data:
+                continue
+
+            if data.get("id") not in instance_ids:
+                continue
+
+            creator_id = data.get("creator_identifier")
+            if not creator_id:
+                continue
+
+            cache.setdefault(creator_id, []).append(node)
+
+    return shared_data
+
+
+class SilhouetteCreator(Creator):
+    default_variants = ['Main']
+    settings_category = "silhouette"
+
+    node_type = "OutputNode"
+
+    @lib.undo_chunk("Create instance")
+    def create(self, product_name, instance_data, pre_create_data):
+
+        session = fx.activeSession()
+        if not session:
+            return
+
+        instance_node = None
+        use_selection = pre_create_data.get("use_selection")
+        selected_nodes = []
+        if use_selection:
+            # Allow to imprint on a currently selected node of the same type
+            # as this creator would generate. If the node is already imprinted
+            # by a Creator then raise an error - otherwise use it as the
+            # instance node.
+            selected_nodes = [
+                node for node in fx.selection() if isinstance(node, fx.Node)]
+            for node in selected_nodes:
+                if node.type == self.node_type:
+                    data = lib.read(node)
+                    if data and data.get("creator_identifier"):
+                        raise CreatorError(
+                            "Selected node is already imprinted by a Creator."
+                        )
+                    instance_node = node
+                    self.log.debug(
+                        f"Using selected node as instance node: {node.label}")
+                    break
+
+        if instance_node is None:
+            # Create new node and place it in the scene
+            instance_node = fx.Node(self.node_type)
+            session.addNode(instance_node)
+            lib.set_new_node_position(instance_node)
+
+            # When generating a new instance node and use selection is enabled,
+            # connect to the first selected node with a matching output type
+            if use_selection and selected_nodes:
+                self._connect_input_to_first_matching_candidate(
+                    instance_node, selected_nodes)
+
+        instance_node.label = session.uniqueLabel(product_name)
+        fx.activate(instance_node)
+
+        # Enforce forward compatibility to avoid the instance to default
+        # to the legacy `AVALON_INSTANCE_ID`
+        instance_data["id"] = AYON_INSTANCE_ID
```

Comment:
You don't have to fill it explicitly, and the comment is irrelevant for new integration.
```suggestion
```

Reply:
Actually are you sure? I've had issues in another integration just recently where without this it'd default to imprinting with `AVALON_INSTANCE_ID` instead. Let me try here.

Reply:
Oh my, if that's the case then we should change it in core...